plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -984,7 +985,12 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, node) {
+	allPods, err := c.podLister.List(labels.Everything())
```

Comment:
So, AFAIU, in `satisfiesExistingPodsAffinity` normal listing is done only when meta is not available but here, we are listing before the call happens to function, so the amortized value may not be equal always.